release-notes-1.2.0.md

```diff
@@ -0,0 +1,763 @@
+
+```ballerina
+# Creates and returns a `Baz` object.
+#
+# # Deprecated
+# This function is deprecated due to undesired side effects since it relies on module level
+# variables. Use the new and improved `newFoo()` function instead.
+@deprecated
+public function foo() returns Baz {
+    // some logic which has side effects
+}
+```
+
+#### Metadata on record and object fields
+
+Metadata (documentation and annotations) are now allowed on record and object fields.
+
+```ballerina
+// An annotation allowed on record fields.
+annotation foo on record field;
+
+// An annotation allowed on object fields.
+annotation bar on object field;
+
+// An annotation allowed on record and object fields.
+annotation baz on field;
+
+# Docs for this record.
+public type Rec record {
+
+    # Docs for this field.
+    @foo
+    @baz
+    string s;
+};
+
+# Docs for this object.
+public type Obj object {
+
+    # Docs for this field.
+    @bar
+    @baz
+    int i = 0;
+};
+```
+
+### Lang Library 
+
+#### Builtin sub types of basic types
+
+Lang Library modules provide a new set of built-in types. Each such built-in type that is a sub type of a basic type `B` is provided by the `lang.B` module. The following are the built-in types provided by the lang library modules.
+
+- Module `ballerina/lang.int`
+  - `Unsigned8`
+  - `Signed8`
+  - `Unsigned16`
+  - `Signed16`
+  - `Unsigned32`
+  - `Signed32`
+
+- Module `ballerina/lang.string`
+  - `Char`
+
+- Module `ballerina/lang.xml`
+  - `Element`
+  - `ProcessingInstruction`
+  - `Comment`
+  - `Text`
+
+```ballerina
+import ballerina/lang.'int;
+import ballerina/lang.'string;  
+
+public function main() {
+    'int:Unsigned8 number = 200;
+    'string:Char char = "A";
+}
+```
+
+#### Redesigned `lang.xml` module
+
+The XML lang module has been revamped to work with the `Element`, `Comment`, `ProcessingInstructions` and `Text` XML built-in types. 
+
+Previously, functions such as `getChildren()` and `getElementName()`, which are specific to an XML element type were allowed to be called on the `xml` type and if they were called on non-element items, it used to result in runtime errors. With `xml` built-in sub type improvements, these functions are statically type checked and only allowed on the built-in sub type `Element`. The same applies to other built-in sub types and functions.
+
+```ballerina
+import ballerina/lang.'xml as xmllib;
+
+public function main() {
+     xmllib:Element element = <xmllib:Element> xml `<elem> hello </elem>`;
+     xml children = element.getChildren();
+   }
+```
+
```

Comment:
```suggestion
- The functions `isElement()`, `isProcessingInstruction()`, `isComment()` and `isText()` have been removed. The same can be achieved using type testing as follows. 
```ballerina
import ballerina/lang.'xml;
public function main() {
   xml cmnt = xml `<!-- hello from comment -->`;
   boolean isComment = cmnt is 'xml:Comment;
}

- The functions `appendChildren()` and `removeChildren()` has been removed.
``` 
```

Reply:
@rdhananjaya please review. anything else we should mention?

Reply:
approved